src/components/ui/ContributorLabel.tsx

```diff
@@ -0,0 +1,59 @@
+import Color from 'color'
+import {Contributor} from 'data/CONTRIBUTORS'
+import PropTypes from 'prop-types'
+import React, {CSSProperties, PureComponent} from 'react'
+import {Label} from 'semantic-ui-react'
+
+interface ContributorLabelProps {
+	contributor: string | Contributor
+	detail?: React.ReactNode
+}
+
+export default class ContributorLabel extends PureComponent<ContributorLabelProps> {
+	static propTypes = {
+		contributor: PropTypes.oneOfType([
+			PropTypes.string,
+			PropTypes.shape({
+				name: PropTypes.string.isRequired,
+				avatar: PropTypes.string,
+			}),
+		]).isRequired,
+		detail: PropTypes.node,
+	}
+
+	normalizeContributor = (contributor: string | Contributor): Contributor => {
+		if (typeof contributor === 'string') {
+			return {
+				name: contributor,
+				jobs: [],
+			}
+		} else {
+			return contributor
+		}
+	}
+
+	render() {
+		const {contributor, detail} = this.props
+
+		const {avatar, name, jobs} = this.normalizeContributor(contributor)
+
+		const style: CSSProperties = {}
+		if (jobs && jobs.length) {
```

Comment:
Would `jobs?.length` not work here?

Reply:
Yes, that should be `jobs?.length`.
I'll fix it up in a sec. Must have just variable replaced it when I was doing that part.